src/extension.ts

```diff
@@ -70,7 +71,9 @@ export async function activate(extensionContext: ExtensionContext): Promise<unkn
     await commands.executeCommand('setContext', 'isVSCode', env.uiKind);
     // UIKind.Desktop ==1 & UIKind.Web ==2. These conditions are checked for browser based & electron based IDE.
     migrateFromOdo018();
+    void extensionContext.globalState.update('hasTekton', await isTektonAware());
```

Comment:
I think we could do this better. Currently this blocks the extension startup, and means you need to restart the extension after you install Tekton, and doesn't take into account switching clusters.

Reply:
These checks run so fast that I think it's worthwhile rerunning it every time we go to run remote deploy.